### ISSUE TITLE
Fully specify Ed25519 in JOSE features

### ISSUE THREAD

**Issue by KendallWeihe** [TBD54566975/web5-rs#128]
Originally brought up here https://github.com/TBD54566975/web5-rs/pull/97#discussion_r1576858213

We need to fully-specifcy Ed25519 rather than EdDSA in our algorithm values

I actually did went ahead and did this [here](https://github.com/TBD54566975/web5-rs/commit/460d8ba70bf8014c6dde13ff875cae79fdb02cfa) but in the process it seems like there are still some unknowns so we can dedicate time to do this right 

@decentralgabe should we still support verification of a JWS Header w/ Ed25519 given the alg value is EdDSA?

Also during the work I realized our did:jwk create function may have room for improvement, we're doing some weird things to use the underlying spruce lib which feels off. Also all of the verification relationships are empty and shouldn't be. Feel free to fix these things here or in a separate work item.

**Comment by decentralgabe** [comment]
> we still support verification of a JWS Header w/ Ed25519 given the alg value is EdDSA?

yes, since what I mentioned is still in draft. the reason I support adopting it immediately is because it reduces confusion. it is sensible to support both as the transition takes place

**Comment by frankhinek** [comment]
Agreed with @decentralgabe 's note above.  Since many libraries won't adopt the new `Ed25519` alg anytime soon we'll need to ensure we at least support both... even if we prefer the more description algorithm identifier.
